packages/vscode-extension/lib/wrapper.js

```diff
@@ -190,53 +221,47 @@ export function AppWrapper({ children, initialProps, ..._rest }) {
             x: frame.left / width,
             y: frame.top / height,
             width: frame.width / width,
-            height: frame.height / height,
+            height: frame.height / height
           };
-          let stackPromise = Promise.resolve(undefined);
-          if (payload.requestStack) {
-            stackPromise = Promise.all(
-              viewData.hierarchy.reverse().map((item) => {
-                const inspectorData = item.getInspectorData((arg) => findNodeHandle(arg));
-                const framePromise = new Promise((resolve, reject) => {
-                  try {
-                    inspectorData.measure((_x, _y, viewWidth, viewHeight, pageX, pageY) => {
-                      resolve({
-                        x: pageX / width,
-                        y: pageY / height,
-                        width: viewWidth / width,
-                        height: viewHeight / height,
-                      });
-                    });
-                  } catch (e) {
-                    reject(e);
-                  }
-                });
-
-                return framePromise
-                  .catch(() => undefined)
-                  .then((frame) => {
-                    return inspectorData.source
-                      ? {
-                          componentName: item.name,
-                          source: {
-                            fileName: inspectorData.source.fileName,
-                            line0Based: inspectorData.source.lineNumber - 1,
-                            column0Based: inspectorData.source.columnNumber - 1,
-                          },
-                          frame,
-                        }
-                      : undefined;
-                  });
-              })
-            ).then((stack) => stack?.filter(Boolean));
-          }
-          stackPromise.then((stack) => {
+          
+          if (!requestStack) {
             devtoolsAgent._bridge.send("RNIDE_inspectData", {
               id: payload.id,
               frame: scaledFrame,
-              stack: stack,
             });
-          });
+            return;
+          }
+
+          const inspectNode = (node, stack) => {
+            // Optimization: we break after reaching fiber node corresponding to OffscreenComponent (with tag 22).
+            if (!node || node.tag === 22) {
+              devtoolsAgent._bridge.send("RNIDE_inspectData", {
+                id: payload.id,
+                frame: scaledFrame,
+                stack: Array.from(stack).filter(Boolean)
+              });
```

Comment:
it'd be easier to follow if we just `return stack` here and move this after `inspectNode` call from line 264. This way it'd be clear that this is a closing statement of the node stack processing logic.

I also added another comment to the recursive call which uses conditional push, therefore eliminating the need to filter here.